pennylane/tape/qnode.py

```diff
@@ -188,18 +193,26 @@ def get_tape(device, interface, diff_method="best"):
         if diff_method == "reversible":
             return QNode._validate_reversible_method(device, interface)
 
+        if diff_method == "adjoint":
+            return QNode._validate_adjoint_method(device, interface)
+
         if diff_method == "device":
             return QNode._validate_device_method(device, interface)
 
         if diff_method == "parameter-shift":
-            return QNode._get_parameter_shift_tape(device), interface, "analytic", device
+            return (
+                QNode._get_parameter_shift_tape(device),
+                interface,
+                device,
+                {"method": "analytic"},

```

Comment:
How come this became a dictionary?

Reply:
It allows us to specify more than one option, for example, `adjoint` is: `{"method": "device", "jacobian_method": "adjoint_jacobian"}` while another device-based Jacobian would be `{"method": "device"}`. We could feasibly have other gradient methods that live on the device in future (if there's a reason to)

Reply:
What's the difference between `method` and `jacobian_method` and why would both be needed?

Reply:
Hmm, imho it was originally meant to describe that a device could have several device jacobians defined, so `method` would indicate that we'd like to use a jacobian defined by the device and then we could select the specific type with `jacobian_method`.

However, the question is still great, would we actually need a separate `jacobian_method`, or could we just go with:
* `method="device"`
* `method="device_adjoint"`?

Guess it comes down to how complex the use case is, i.e., would we expect to see (many) devices that are different in their default `device` diff method> If it turns out that the use cases are more complex, then it can be worth having these two entries.